drivers/potentiometer/common/dpot.c

```diff
@@ -0,0 +1,433 @@
+/*******************************************************************************
+ *   @file   dpot.c
+ *   @brief  Source file for the digital potentiometer drivers
+********************************************************************************
+ * Copyright 2025(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include "dpot.h"
+
+/**
+ * @brief Initialize the digital potentiometer.
+ * @param param - digipot init parameters.
+ * @param desc - digipot descriptor.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int dpot_init(DPOT_INIT_PARAM *pdpot_init_param, struct dpot_dev **desc)
+{
+	int ret;
+	enum dpot_dev_id eDevice;
+	eDevice = pdpot_init_param->device_id;
+
+	struct dpot_init_param *param = &dpot_info[eDevice].dpot_init_params;
```

Comment:
`dpot_info` is a global, the point of this API is to be self sufficient and not expect users to initialize globals for the API to behave correctly... Everything must go through init params.